### PR TITLE
Chime4  Modified

### PR DESCRIPTION
**Pull request by Szu-JuiChen**
I updated the baseline and fix few bugs for running the scripts. I also modified run.sh for easy use the new version.

----update----
Modified as suggestion. 10/21 1:00 pm

### CODE REVIEW

File: egs/chime3/s5/run.sh [hunk 1]
```diff
@@ -29,6 +29,11 @@ set -o pipefail
 # chime3_data=`pwd`/../..
 # Otherwise, please specify it, e.g.,
 chime3_data=/data2/archive/speech-db/original/public/CHiME3
+
+case $(hostname -d) in
```
Comment (danpovey) [inline egs/chime3/s5/run.sh]: this is not portable, do
`case $(hostname) in   *.clsp.jhu.edu)`

File: egs/chime4/s5_1ch/run.sh [hunk 2]
```diff
@@ -28,15 +39,41 @@ set -o pipefail
 chime4_data=`pwd`/../..
 # Otherwise, please specify it, e.g.,
 chime4_data=/db/laputa1/data/processed/public/CHiME4
+
+case $(hostname -d) in
```
Comment (danpovey) [inline egs/chime4/s5_1ch/run.sh]: here too.

File: egs/chime4/s5_1ch/run.sh [hunk 3]
```diff
@@ -7,11 +7,22 @@
...
+# Usage: 
+# For using original baseline, set baseline to chime4_official. In this case,
+# you don't need to run flatstart since we already provided the trained model in 
+# tools/ASR_models.
+# For advanced baseline, please run flatstart first to get the models.
+# By default, we assume you already got the models.
 
 # Config:
 stage=0 # resume training with --stage=N
+baseline={"chime4_official","advanced"}
```
Comment (danpovey) [inline egs/chime4/s5_1ch/run.sh]: I don't know what you are trying to do here. I think maybe what you mean is something like this:


```
baseline=advanced
# By default this runs the 'advanced' baseline.  Uncomment the following to run
# the 'chime4_official' (i.e. less advanced) baseline.
# baseline=chime4_official
#

```

File: egs/chime4/s5_2ch/run.sh [hunk 4]
```diff
@@ -7,11 +7,22 @@
 #                Mitsubishi Electric Research Labs (Shinji Watanabe)
 #  Apache 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
 
+# Added advanced baseline
```
Comment (danpovey) [inline egs/chime4/s5_2ch/run.sh]: this is not the place for a changelog: add your name above, e.g. after watanabe, say:
`#           2017   Szu-Jui Chen `
and remove this.

File: egs/chime4/s5_2ch/run.sh [hunk 5]
```diff
@@ -7,11 +7,22 @@
...
+
 . ./path.sh
 . ./cmd.sh
+#####Baseline settings#####
+# Usage: 
+# For using original baseline, set baseline to chime4_official. In this case,
+# you don't need to run flatstart since we already provided the trained model in 
+# tools/ASR_models.
+# For advanced baseline, please run flatstart first to get the models.
+# By default, we assume you already got the models.
```
Comment (danpovey) [inline egs/chime4/s5_2ch/run.sh]: can you please be more explicit here about what the user is supposed to run, e.g. say what they should run first and what they should run next.

File: egs/chime4/s5_2ch/path.sh [hunk 6]
```diff
@@ -1,6 +1,8 @@
 export KALDI_ROOT=`pwd`/../../..
+[ -f $KALDI_ROOT/tools/extras/env.sh ] && . $KALDI_ROOT/tools/extras/env.sh
 [ -f $KALDI_ROOT/tools/env.sh ] && . $KALDI_ROOT/tools/env.sh
 export PATH=$PWD/utils/:$KALDI_ROOT/tools/openfst/bin:$PWD:$PATH
 [ ! -f $KALDI_ROOT/tools/config/common_path.sh ] && echo >&2 "The standard file $KALDI_ROOT/tools/config/common_path.sh is not present -> Exit!" && exit 1
 . $KALDI_ROOT/tools/config/common_path.sh
 export LC_ALL=C
+export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/home/dpovey/libs/
```
Comment (danpovey) [inline egs/chime4/s5_2ch/path.sh]: this is a local thing that doesn't belong here.

Comment (Szu-JuiChen) [reply egs/chime4/s5_2ch/path.sh]: oh! Sorry. I do not know about this.

File: egs/chime4/s5_2ch/run.sh (hunk 5 above)
Comment (Szu-JuiChen) [reply egs/chime4/s5_2ch/run.sh]: Sure. I will rewrite this part.

**Comment by danpovey** [comment]
@sw005320, does this look OK to you?

**Comment by sw005320** [comment]
Yes, this is OK.
I already read his code, and he included my comments in the pull-request version.
This method (using all 6 channels for training) has been used for most of CHiME-4 top submissions, and would be a good advanced baseline.

**Pull request merged**